main.js

```diff
@@ -0,0 +1,147 @@
+var VideoModel = Backbone.Model.extend({
+  defaults: {
+    videoId: '',
+    title: '',
+    description: '',
+    thumbnails: '',
+  },
+});
+
+var AppModel = Backbone.Model.extend({
+  defaults: function () {
+    return {
+      videos: new VideosCollection(),
+      searchTerm: '',
+      currentVideo: null,
+      apiKey: 'AIzaSyCXJ4dabPUSUUVCbFRMWJP59dgxIU28pOM',
+    };
+  },
+
+  initialize: function () {
+    //default search
+    this.set('searchTerm', 'babies laughing at dogs'); 
+    this.updateSearch(); 
+  }, 
+
+  updateSearch: function () {
+    this.get('videos').makeFetch(this.get('searchTerm'), this.get('apiKey'));
+  },
+  
+  updateCurrentVideo: function (videoId) {
+    var allVideos = this.get('videos');
+    var currentVideo = allVideos.findWhere({videoId: videoId});
+
+    this.set('currentVideo', currentVideo);
+  },
+});
+
+var VideosCollection = Backbone.Collection.extend({
+  url:  '',
+
+  model: VideoModel,
+
+  makeFetch: function (searchTerm, apiKey) {
+    this.url = 'https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=5&q=' + searchTerm + '&type=video&videoEmbeddable=true&key=' + apiKey;
+
+    this.fetch(); 
+  },
+
+  parse: function(response) {
+    var resVideo = response.items;
+
+    return resVideo.map(function (video) {
+      return {
+        videoId: video.id.videoId,
+        title: video.snippet.title,
+        description: video.snippet.description,
+        thumbnails: video.snippet.thumbnails.default.url, 
+      }
+    });
+  },
+  //sets a main video at random 
+  defaultVideo: function () {
+    var randomVideo = Math.floor(Math.random() * this.length);
+    return this.at(randomVideo); 
+  },
+});
+
+var VideoView = Backbone.View.extend({
+  className: 'side-bar-videos',
+
+  template: Handlebars.compile($('#side-bar-template').html()),
+
+  render: function () {
+    this.$el.html(this.template(this.model.toJSON())); 
+    return this; 
+  },
+});
+
+var MainVideoView = Backbone.View.extend({
+  className: 'main-video',
+
+  template: Handlebars.compile($('#main-video-template').html()),
+
+  render: function () {
+    this.$el.html(this.template(this.model.toJSON())); 
+    return this; 
+  },
+});
+
+var AppView = Backbone.View.extend({
+  el: $('body'),
+
+  events: {
+    'click .search-button': 'handleSearchButtonClick',
+    'click .video-thumbnails': 'getCurrentVideoId',
+  },
+
+  initialize: function () {
+    this.$searchTerm = this.$('#video-search'); 
+    this.$mainVideo = this.$('.main');
+    this.$sideBar = this.$('.side-bar');
+    
+    this.listenTo(this.model.get('videos'), 'add', this.renderVideos);
```

Comment:
This function will get called once for every video that comes back - for this reason I usually prefer to use `reset`